src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,296 @@
+            auto robot_pos = event.common.robot.position();
+
+            if ((ball.position() - robot_pos).length() >
+                DIST_TO_FRONT_OF_ROBOT_METERS + 2 * BALL_MAX_RADIUS_METERS)
+            {
+                Point ball_receive_pos = ball.position();
+
+                if (ball.velocity().length() != 0)
+                {
+                    ball_receive_pos = closestPoint(
+                        robot_pos,
+                        Line(ball.position(), ball.position() + ball.velocity()));
+                }
+
+                Angle ball_receive_orientation =
+                    (ball.position() - robot_pos).orientation();
+
+                event.common.set_intent(std::make_unique<MoveIntent>(
+                    event.common.robot.id(), ball_receive_pos, ball_receive_orientation,
+                    0, DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                    AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                    MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+            }
+        };
+
+        /**
+         * Check if the pass has started by checking if the ball is moving faster
+         * than a minimum speed.
+         *
+         * @param event ReceiverFSM::Update event
+         * @return true if the pass has started
+         */
+        const auto pass_started = [](auto event) {
+            return event.common.world.ball().velocity().length() > MIN_PASS_START_SPEED;
+        };
+
+        /**
+         * Check if the pass has finished by checking if we the robot has
+         * a ball near its dribbler.
+         *
+         * @param event ReceiverFSM::Update event
+         * @return true if the ball is near a robots mouth
+         */
+        const auto pass_finished = [](auto event) {
+            // We tolerate imperfect passes that hit the edges of the robot,
+            // so that we can quickly transition out and grab the ball.
+            return event.common.robot.isNearDribbler(event.common.world.ball().position(),
+                                                     ROBOT_MAX_RADIUS_METERS);
+        };
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *undecided_s + update_e[onetouch_possible] / update_onetouch = onetouch_s,
+            undecided_s + update_e[!onetouch_possible] / update_receive  = receive_s,
+            receive_s + update_e[!pass_started] / update_receive         = receive_s,
+            onetouch_s + update_e[!pass_started] / update_onetouch       = onetouch_s,
+            receive_s + update_e[pass_started && !pass_finished] / adjust_receive =
+                receive_s,
+            onetouch_s + update_e[pass_started && !pass_finished] / update_onetouch =
+                onetouch_s,
+            receive_s + update_e[pass_started && pass_finished] / update_receive   = X,
+            onetouch_s + update_e[pass_started && pass_finished] / update_onetouch = X);
```

Comment:
couldn't we replace `pass_started && pass_finished` with just `pass_finished`?

Reply:
yup, replaced